components/num-util/src/fixed_decimal.rs

```diff
@@ -385,15 +385,14 @@ impl FromStr for FixedDecimal {
         if input_str == "" || input_str == "-" {
             return Err(Error::Syntax);
         }
-        let no_sign_str: &str;
         let mut is_negative = false;
-        if &input_str[0..1] == "-" {
+        let input_str = input_str.as_bytes();
+        let mut no_sign_str = input_str;
+        if input_str[0] == b'-' {
             is_negative = true;
             no_sign_str = &input_str[1..];
-        } else {
-            no_sign_str = input_str;
         }
-
+        let no_sign_str = no_sign_str;
```

Comment:
?

Reply:
Turns out removing mutability from `no_sign_str` actually matters. 
The improvement is so tiny that I don't really know, from measurement alone, whether it's real.

Reply:
I suspect red herring. Mutability doesn't produce any chance in assembly. It only affects whether the MIR gets produced or rejected much earlier in the process.

Reply:
Done.